thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2911,10 +3055,17 @@ class HashDedupSlaveActivityBase : public CSlaveActivity
         ActivityTimer t(totalCycles, timeActivities);
         if (eos)
             return NULL;
+        if (keepBestRowsReadyToReturn)
+        {
+            if (bucketHandlerIterator->next())
+                return bucketHandlerIterator->query();
```

Comment:
This should be returning a linked row - I'm not sure why you don't get a heap fault.